typescriptFixme.ts

```diff
@@ -1,115 +1,143 @@
   return i * i;
 };
 
 //change f4 to accept a string or a number as a parameter
-function f4(i) {
+function f4(i: number|string) {
   console.log("The value was " + i);
 }
 
 // Annotate the person objects with an interface. Some properties are optional
+interface Person {
+  name: string;
+  age?: number;
+  move: () => void;
+}
+
 var person1 = { name: "Bobby", move: function () {} };
-var person2 = { name: "Bobby", age: 42, move: function () {} };
+var person2 = { name: "Bobby", age: 42, move: function () { } };
+
 
 // Annotate mySearch to describe a function that accepts two string parameters
-var mySearch = function (src, sub) {
+var mySearch: (src: string, sub: string) => boolean = function (src, sub) {
   return src.search(sub) != -1;
 };
-
 //annotate the constructor arguments and properties
 class Point {
-  x = 0;
-  constructor(x, y = 0) {
+  x: number = 0;
+  y: number = 0;
+
+  constructor(x: number, y: number = 0) {
     this.x = x;
     this.y = y;
   }
-  dist() {
+
+  dist(): number {
     return Math.sqrt(this.x * this.x + this.y * this.y);
   }
 }
 
+
 // annotate `PointPerson` to use all properties of `Person` and then add the missing required property
-class PointPerson {
-  constructor(name) {
+class PointPerson implements Person {
+  name: string;
+  age?: number;
+  move(): void {}
+  constructor(name: string) {
     this.name = name;
   }
 }
 
 // Inheritance
 // Make `Point3D` inherit from `Point`
 // Don't forget to call the parent constructor
-class Point3D {
-  constructor(x, y, z = 0) {
+class Point3D extends Point {
+  z: number;
+  constructor(x: number, y: number, z: number = 0) {
+    super(x, y);
     this.z = z;
   }
-  dist() {
+  dist(): number {
     var d = super.dist();
     return Math.sqrt(d * d + this.z * this.z);
   }
 }
 
 // Make `Tuple` Generic that it accepts to generic types as arguments
 
-class Tuple {
-  constructor(item1, item2) {
+class Tuple<T, U> {
+  item1: T;
+  item2: U;
+  constructor(item1: T, item2: U) {
     this.item1 = item1;
     this.item2 = item2;
   }
 }
 
 // create a generic interface called Pair that accepts one generic type for two properties named item1 and item2
 
+interface Pair<T> {
+  item1: T;
+  item2: T;
+}
+
 // create a function that will take a pair with a generic argument and return a `Tuple`
-function pairToTuple(pair) {
+function pairToTuple<T>(pair: Pair<T>): Tuple<T, T> {
   return new Tuple(pair.item1, pair.item2);
 }
 
 var tuple = pairToTuple({ item1: "hello", item2: "world" });
 
 // annotate `make`, `model`, and `year` to be not mutable (read only) outside the constructor
 class Car {
-  make;
-  model;
-  year = 2018;
+  readonly make: string;
+  readonly model: string;
+  readonly year: number;
 
   constructor() {
     this.make = "Unknown Make";
     this.model = "Unknown Model";
+    this.year = 2018;
   }
 }
 
+
 // annotate this array
-var arrayOfAnyType = [1, "string", false];
+var arrayOfAnyType: any[] = [1, "string", false];
```

Comment:
don't use `any`. Change this to be an array that can accept all the three types used using union